### PR TITLE
Adds a searchable map to upload page

### PR DESCRIPTION
**Pull request by michelleli18**
There is now a map that can be searched for locations on upload page. Can navigate between upload page and main page with buttons now. This commit does not include javascript or servlet parts of the page yet, those are still work in progress. Just pushing so that everyone gets an updated page to better work with!

### CODE REVIEW

File: project/src/main/java/com/google/sps/servlets/UploadReport.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+
+
+
+@WebServlet("/upload")
+@MultipartConfig
+// @Override
```
Comment (EvanJuV) [inline project/src/main/java/com/google/sps/servlets/UploadReport.java]: Do we want to Override or not? If we don't need it anymore, please delete it.

File: project/src/main/webapp/upload-page/upload.html [hunk 2]
```diff
@@ -5,25 +5,103 @@
...
             <input type="submit" value="Return to Main Page" />
         </form>
-        <center>
-            <h1>Upload a report!</h1>
-            <form action="/report-handler" method="POST" enctype="multipart/form-data">
-            <!-- <form action="/report-upload" method="POST" enctype="multipart/form-data"></form> -->
-                <p>Please upload a quick description of the collision:</p>
-                <textarea name="text-input">E.g. date, time-of-day, car types of those involved, etc.</textarea>
+        <center><h1>Upload a report!</h1></center>
+<<<<<<< HEAD
```
Comment (EvanJuV) [inline project/src/main/webapp/upload-page/upload.html]: Please resolve this diff by selecting the HTML sections that you want to keep. The section between the <<< HEAD and === markers are your local changes, and the section between === and >>>> ec2c0ce4d528... is how this same section looks in the commit your merging.

File: project/src/main/webapp/upload-page/upload.html [hunk 3]
```diff
@@ -5,25 +5,103 @@
...
+                <li id = "inline-textbox">Time of incident:</li> <input type="datetime-local">
+                <br/>
+                <li id = "inline-textbox">Description of the collision:</p> <textarea name="description">E.g. car types of those involved, etc.</textarea>
+>>>>>>> ec2c0ce4d528d034d32b449e10a0a7424d0c8ba8
+                <br/>
+            </ul>
+            <br/>
+            <p>Please attach a photo with your description if you can:</p>
+            <input type="file" name="image">
+<<<<<<< HEAD
```
Comment (EvanJuV) [inline project/src/main/webapp/upload-page/upload.html]: Please resolve this other diff.

File: project/src/main/webapp/upload-page/upload_script.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+<<<<<<< HEAD
```
Comment (EvanJuV) [inline project/src/main/webapp/upload-page/upload_script.js]: Resolve this diff.

File: project/src/main/webapp/upload-page/upload_script.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+      // For each place, get the icon, name and location.
+      const bounds = new google.maps.LatLngBounds();
+  
+      places.forEach((place) => {
+        if (!place.geometry || !place.geometry.location) {
+          console.log("Returned place contains no geometry");
+          return;
+        }
+  
+<<<<<<< HEAD
```
Comment (EvanJuV) [inline project/src/main/webapp/upload-page/upload_script.js]: Resolve this other diff.

File: project/src/main/webapp/upload-page/upload_script.js [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+          anchor: new google.maps.Point(17, 34),
+          scaledSize: new google.maps.Size(25, 25),
+        };
+  
+>>>>>>> ec2c0ce4d528d034d32b449e10a0a7424d0c8ba8
+        // Create a marker for each place.
+        markers.push(
+          new google.maps.Marker({
+            map,
+<<<<<<< HEAD
```
Comment (EvanJuV) [inline project/src/main/webapp/upload-page/upload_script.js]: This one as well.

File: project/src/main/webapp/upload-page/upload_script.js [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+        if (place.geometry.viewport) {
+          // Only geocodes have viewport.
+          bounds.union(place.geometry.viewport);
+        } else {
+          bounds.extend(place.geometry.location);
+        }
+      });
+      map.fitBounds(bounds);
+    });
+<<<<<<< HEAD
```
Comment (EvanJuV) [inline project/src/main/webapp/upload-page/upload_script.js]: Also this one please.

File: project/src/main/webapp/upload-page/upload_style.css [hunk 8]
```diff
@@ -10,16 +11,117 @@
...
+    color: #fff;
+    background-color: #4d90fe;
+    font-size: 25px;
+    font-weight: 500;
+    padding: 6px 12px;
+  }
+  
+  #target {
+    width: 345px;
+<<<<<<< HEAD
```
Comment (EvanJuV) [inline project/src/main/webapp/upload-page/upload_style.css]: And finally this one.

**Review by EvanJuV — CHANGES_REQUESTED** [review]

**Comment by michelleli18** [comment]
Updated with the newest commit just now! It resolved all the diffs and also took care of the required fields for the survey

File: project/src/main/webapp/upload-page/upload.html [hunk 9]
```diff
@@ -5,25 +5,65 @@
     <title>Report a Case</title>
     <link rel="stylesheet" href="upload_style.css">
     <script src="upload_script.js"></script>
+    <script src="https://polyfill.io/v3/polyfill.min.js?features=default"></script>
+
   </head>
   <body>
       <div id="content"> 
-        <form action="index.html">
+        <form action="../index.html">
```
Comment (EvanJuV) [inline project/src/main/webapp/upload-page/upload.html]: I think you can just do "/index.html" as that's the absolute path of the index.

File: project/src/main/webapp/upload-page/upload.html [hunk 10]
```diff
@@ -5,25 +5,65 @@
...
-            <form action="/report-handler" method="POST" enctype="multipart/form-data">
-            <!-- <form action="/report-upload" method="POST" enctype="multipart/form-data"></form> -->
-                <p>Please upload a quick description of the collision:</p>
-                <textarea name="text-input">E.g. date, time-of-day, car types of those involved, etc.</textarea>
+        <center><h1>Upload a report!</h1></center>
+        <form action="/upload" method="POST" enctype="multipart/form-data">
+        <!-- contact info, title, latitude, longitude, date+time (use html component), img url -->
+            <p>Please complete the following details of the report:</p>
+            <ul>
+                <li id = "inline-textbox">
```
Comment (EvanJuV) [inline project/src/main/webapp/upload-page/upload.html]: I see this id="inline-textbox" being used more than once. In CSS, IDs are used to target a single specific element. If you want to target multiple elements use a class or change directly the element type.

Also please remove the spaces before and after =. 

Apply changes to any other li element.

**Review by EvanJuV — CHANGES_REQUESTED** [review]

**Comment by chanjonathan** [comment]
can we merge this to main?

**Review by EvanJuV — APPROVED** [review]
LGTM

**Pull request merged**